src/Node.php

```diff
@@ -15,11 +15,9 @@ class Node implements \JsonSerializable {
     /** @var Node | null */
     public $parent;
 
-    public function getKind() {
-        // TODO remove all references to getKind
-        $constants = (new \ReflectionClass("Microsoft\\PhpParser\\NodeKind"))->getConstants();
-        $nameParts = explode("\\",get_class($this));
-        return $constants[end($nameParts)];
+    public function getNodeKindName() : string {
+        $parts = explode('\\', static::class);
+        return $parts[count($parts) - 1];
```

Comment:
An alternative implementation would be to use strrpos(), which avoid creating a temporary array.